ansible-scylla-node/tasks/ssl.yml

```diff
@@ -25,59 +69,61 @@
+        path: "{{ scylla_ssl.localhost_cert_path | default(inventory_dir) }}/ssl/{{ inventory_hostname }}"
         state: directory
       delegate_to: localhost
 
     - name: Generate an OpenSSL private key.
       openssl_privatekey:
-        path: "./ssl/{{ inventory_hostname }}/{{ inventory_hostname }}.pem"
+        path: "{{ scylla_ssl.localhost_cert_path | default(inventory_dir) }}/ssl/{{ inventory_hostname }}/{{ inventory_hostname }}.pem"
       delegate_to: localhost
 
     - name: Generate an OpenSSL Certificate Signing Request
       openssl_csr:
-        path: "./ssl/{{ inventory_hostname }}/{{ inventory_hostname }}.csr"
-        privatekey_path: "./ssl/{{ inventory_hostname }}/{{ inventory_hostname }}.pem"
+        path: "{{ scylla_ssl.localhost_cert_path | default(inventory_dir) }}/ssl/{{ inventory_hostname }}/{{ inventory_hostname }}.csr"
+        privatekey_path: "{{ scylla_ssl.localhost_cert_path | default(inventory_dir) }}/ssl/{{ inventory_hostname }}/{{ inventory_hostname }}.pem"
         common_name: "{{ inventory_hostname }}.{{ scylla_cluster_name }}.internal"
       delegate_to: localhost
 
     - name: Generate an OpenSSL certificate signed with our CA certificate
       openssl_certificate:
-        path: "./ssl/{{ inventory_hostname }}/{{ inventory_hostname }}.crt"
-        csr_path: "./ssl/{{ inventory_hostname }}/{{ inventory_hostname }}.csr"
-        ownca_path: "./ssl/ca/{{scylla_cluster_name }}-ca.crt"
-        ownca_privatekey_path: "./ssl/ca/{{ scylla_cluster_name }}-ca.pem"
+        path: "{{ scylla_ssl.localhost_cert_path | default(inventory_dir) }}/ssl/{{ inventory_hostname }}/{{ inventory_hostname }}.crt"
+        csr_path: "{{ scylla_ssl.localhost_cert_path | default(inventory_dir) }}/ssl/{{ inventory_hostname }}/{{ inventory_hostname }}.csr"
+        ownca_path: "{{ scylla_ssl.localhost_cert_path | default(inventory_dir) }}/ssl/ca/{{scylla_cluster_name }}-ca.crt"
+        ownca_privatekey_path: "{{ scylla_ssl.localhost_cert_path | default(inventory_dir) }}/ssl/ca/{{ scylla_cluster_name }}-ca.pem"
         provider: ownca
       delegate_to: localhost
+  when: _crt_and_key_exist == False
 
-    - name: Copy the certificates into their proper locations
-      copy:
-        src: "{{ item }}"
-        dest: "{{ scylla_ssl.cert_path }}/{{ item | basename }}"
-        owner: root
-        group: root
-        mode: '0644'
-      become: true
-      loop:
-        - "./ssl/ca/{{ scylla_cluster_name }}-ca.crt"
-        - "./ssl/{{ inventory_hostname }}/{{ inventory_hostname }}.crt"
-        - "./ssl/{{ inventory_hostname }}/{{ inventory_hostname }}.pem"
+- name: Copy the certificates into their proper locations
+  copy:
+    src: "{{ item }}"
+    dest: "{{ scylla_ssl.cert_path }}/{{ item | basename }}"
+    owner: root
+    group: root
+    mode: '0644'
+  become: true
+  loop:
+    - "{{ scylla_ssl.localhost_cert_path | default(inventory_dir) }}/ssl/ca/{{ scylla_cluster_name }}-ca.crt"
```

Comment:
Doesn't it feel wrong to you that you call `default` 4 (or more?) times for the same variable? Why not resolving it into a fact or a variable once and then just use it?

Reply:
Fixed